File: Bitwise/Easy/1342. Number of Steps to Reduce a Number to Zero.cpp
class Solution {
public:
    int numberOfSteps(int num) {
        int ans = help(num,0);
        return ans;
    }
    int help(int n,int cnt) {
        if(n==0) return cnt;
        if(n%2==0) return help(n/2,cnt+1);
        else return help(n-1,cnt+1);
    }
};

File: Recursion/Easy/1342. Number of Steps to Reduce a Number to Zero.java
class Solution {
    public int numberOfSteps(int num) {
        int ans = func(num);
        return ans;
    }
    static int func(int n) {
        if(n==0) return 0;
        if(n%2==0) return 1+func(n/2);
        else return 1+func(n-1);
    }
}